src/llmcompressor/modifiers/smoothquant/base.py

```diff
@@ -196,31 +192,38 @@ def _resolve_mappings(self, model: Module) -> List[SmoothQuantMapping]:
         Transforms the list of activations to smooth and their corresponding weights
         into SmoothQuantMapping objects, resolving regular expressions.
 
-        For each activation in the mapping list, we find the corresponding weight to
-        balance by searching for the longest substring. For instance, if our balance
-        weight is ".*re:.*q_proj" and the activation is "re:.*self_attn_layer_norm" we
-        would match model.layer.0.p_proj to model.layer.0.self_attn_layer_norm and
-        repeat for model.layer.1 and so on
+        For each activation in the mapping list, we find ALL corresponding weights to
+        balance by matching within the parent scope. This ensures all matching layers
+        are included, which is critical for MoE models where multiple experts need to
+        be balanced.
         """
         resolved_mappings = []
         for to_balance, to_smooth in self.mappings:
-            to_smooth_layers = get_layers(to_smooth, model)
-            for layer_name, smooth_layer in to_smooth_layers.items():
-                if not match_targets(layer_name, self.ignore)[0]:
-                    balance_layers = []
-                    for balance_suffix in to_balance:
-                        # find the submodule that matches the activation layer
-                        _, balance_layer = get_matching_layer(
-                            balance_suffix, layer_name, model
-                        )
-                        if balance_layer:
-                            balance_layers.append(balance_layer)
-                    # each mapping can contain multiple layers to balance, but only
-                    # one layer to smooth
-                    mapping = SmoothQuantMapping(
-                        layer_name, smooth_layer, balance_layers
+            to_smooth_list = [to_smooth] if isinstance(to_smooth, str) else to_smooth
+
+            for smooth_name, smooth_layer in match_named_modules(
+                model, to_smooth_list, self.ignore
+            ):
+                # Search for balance layers within the parent scope
+                smooth_parent_name = ".".join(smooth_name.split(".")[:-1])
+                smooth_parent = (
+                    get_layer_by_name(smooth_parent_name, model)
+                    if smooth_parent_name
+                    else model
+                )
+
+                balance_layers = []
+                for balance_regex in to_balance:
+                    for _, balance_layer in match_named_modules(
+                        smooth_parent, [balance_regex], self.ignore
+                    ):
+                        balance_layers.append(balance_layer)
```

Comment:
I think we can just do 
```python
for _, balance_layer in match_named_modules(
    smooth_parent, to_balance, self.ignore
):
    balance_layers.append(balance_layer)
```
since `match_named_modules` already contains a loop over targets ([link](https://github.com/vllm-project/compressed-tensors/blob/73c2cf935b53e0078be7766c5ee064755d980d78/src/compressed_tensors/utils/match.py#L65)).

or maybe even
```python
balance_layers = [balance_layer for _, balance_layer in match_named_modules(smooth_parent, to_balance, self.ignore)]

```

Reply:
Accepted! see comment